File: 5_Longest Palindromic Substring/Dennaolan.cpp
class Solution {
public:
    string longestPalindrome(string s) {
      int l, r;
      bool enter;
      string ans;
        
      for ( int i = 0 ; i < s.size() ; i++ ) {
        enter = false;
        for ( l = i-1, r = i+1 ; l >= 0 && r < s.size() ; l--, r++ ) {
          enter = true;
          if ( s[l] != s[r] )
            break;
        }
          
        l++;
        r--;

        if ( !enter || l == r ) {
          if ( ans.size() == 0 )
            ans = s[i];
        }
        else
          ans = s.substr( l, r-l+1 ).size() > ans.size() ? s.substr( l, r-l+1 ) : ans;
      }
      
      for ( int i = 0 ; i < s.size() ; i++ ) {
        enter = false;
        for ( l = i, r = i+1 ; l >= 0 && r < s.size() ; l--, r++ ) {
          enter = true;
          if ( s[l] != s[r] )
            break;
        }
          
        l++;
        r--;
        
        if ( !enter || l > r ) {
          if ( ans.size() == 0 )
            ans = s[i];
        }
        else
          ans = s.substr( l, r-l+1 ).size() > ans.size() ? s.substr( l, r-l+1 ) : ans;
      }

      return ans;
    }
      
    
private:

};

File: 50_Pow(x, n)/Dennaolan.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        ans = 1
        loop_n = abs( n )
        
        if n == 0:
            return ans
        else:
            while loop_n > 0:
                if loop_n % 2 == 0:
                    ans = ans
                else:
                    ans = ans * x
                x = x * x   
                loop_n = int( loop_n / 2 )

            if n > 0:
                return ans
            else:
                return 1/ans
